### ISSUE TITLE
no way to specify pre-existing file

### ISSUE THREAD

**Issue by romaninsh** [atk4/ui#356]
Current implementation of Upload Field decorator does not have a way to specify a pre-existing file. For example if user is editing record that have already file uploaded, it is not displayed.

Here is recommended implementation:

``` php
$form->add('file', ['Upload']);
$form->set('file', $token);
```

This should display "token" as a file-name and allow user to press "x" which would also call onDelete().

Because sometimes token is not same as file name, there should be a way to override filename:

``` php
$field = $form->add('file', ['Upload']);
$form->set('file', $token);

$field->originalFilename = 'myfile.txt';
```

For images you can additionally specify thumbnail:

``` php
$field->thumbnail = 'thumb.png';  // url
```

I need this for https://github.com/atk4/filestore

**Comment by ibelar** [comment]
is $form-set('file', $token) the correct implementation? 
Should it be $fieldname->set($token) instead.

When trying to do $form->set('field_name', 'value') on any fields, it throw an error.

I can easily update the js uploadFile plugin using:

$file = $form->addField('file')
$file->set($token);

**Comment by romaninsh** [comment]
yes, that would be great. I'll check why $form->set is not working, perhaps it should be $form->model->set('field_name')..

**Comment by PhilippGrashoff** [comment]
Seems this was fixed by #360, can it be closed?
